src/components/Modals/ContactCardModal.jsx

```diff
@@ -1,96 +1,85 @@
-    const { contact, deleteContact, onDeleteContact, onClose } = this.props
+  const deleteContact = async (contactParam = null) => {
+    const { contact, deleteContact, onDeleteContact, onClose } = props
     onClose && onClose()
     await deleteContact(contactParam ? contactParam : contact)
     onDeleteContact && onDeleteContact(contactParam ? contactParam : contact)
   }
 
-  toggleEditMode = () => {
-    this.setState(state => ({
-      editMode: !state.editMode
-    }))
+  const toggleEditMode = () => {
+    setEditMode(!editMode)
   }
 
-  render() {
-    const { onClose, t, id } = this.props
-    const { editMode, shouldDisplayConfirmDeleteModal } = this.state
+  const { onClose, t, id } = props
 
-    return (
-      <Modal into="body" dismissAction={onClose} size="xlarge" mobileFullscreen>
-        <Query query={client => client.get(DOCTYPE_CONTACTS, id)}>
-          {({ data: contact, fetchStatus: fetchContactStatus }) => {
-            return (
-              <Query
-                query={client =>
-                  client
-                    .find('io.cozy.contacts.groups')
-                    .where({
-                      trashed: { $exists: false }
-                    })
-                    .sortBy([{ name: 'asc' }])
-                    .indexFields(['name'])
-                }
-              >
-                {({ data: allGroups, fetchStatus: allGroupsContactStatus }) => {
-                  if (
-                    fetchContactStatus !== 'loaded' ||
-                    allGroupsContactStatus !== 'loaded'
-                  ) {
-                    return <SpinnerContact size="xxlarge" />
-                  }
-                  return (
-                    <DumbContactCardModal
-                      editMode={editMode}
-                      contact={contact}
-                      allGroups={allGroups}
-                      t={t}
-                      toggleConfirmDeleteModal={this.toggleConfirmDeleteModal}
-                      toggleEditMode={this.toggleEditMode}
-                      shouldDisplayConfirmDeleteModal={
-                        shouldDisplayConfirmDeleteModal
-                      }
-                      deleteContact={this.deleteContact}
-                    />
-                  )
-                }}
-              </Query>
-            )
-          }}
-        </Query>
-      </Modal>
-    )
-  }
+  const resultContactById = useQuery(
+    queryContactById(id).definition,
+    queryContactById(id).options
+  )
+  const resultAllGroups = useQuery(
+    queryAllGroups.definition,
+    queryAllGroups.options
+  )
+
+  const showResult =
+    (!isCollectionLoading(resultContactById) ||
+      isCollectionInCache(resultContactById)) &&
+    (!isCollectionLoading(resultAllGroups) ||
+      isCollectionInCache(resultAllGroups))
+
+  const contact =
+    resultContactById.data !== null ? resultContactById.data[0] : {}
+  const allGroups = resultAllGroups.data !== null ? resultAllGroups.data : []
```

Comment:
not sure we need this `resultAllGroups.data !== null` anymore since we use `allGroups` only when `showResult` is true === data loaded, no ? Same for `resultContactById` ?

Reply:
We can simplify by using `showResult` but at the first render the data are still null. But we can remove those const and use the data directly in the return part (wrap into `showResult`)

Reply:
You right, the data will be null at the first render. But we don't use them at this time, no? We only read them if `showResult` (or `dataHaveBeenLoaded`) is true, right? So we don't need to deal with `null` data, right?

Reply:
We only read them if showResult (or dataHaveBeenLoaded) is true, right? > Yes. Code updated according to `use the data directly in the return part`. What do you think ?

Reply:
Better! But I'm not very convinced by reading `resultContactById.data[0]`. Are you sure that getById() returns an array and not the object? 🤔

Reply:
Yes... It will not work otherwise... Besides, browser returns this
```
Failed prop type: Invalid prop `contact` of type `array` supplied to `ContactCard`, expected `object`.
```

Reply:
I think that Q().getById() should returns the related document, not an array containing one document. It should behave like the DocumentCollection.get(), no? If yes, we should open an issue on cozy-client

Reply:
issue : https://github.com/cozy/cozy-client/issues/764